onnxruntime/core/framework/sparse_cooformat_rep.cc

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/framework/sparse_cooformat_rep.h"
+#include "core/framework/data_transfer_manager.h"
+
+namespace onnxruntime {
+SparseCooFormatRep::~SparseCooFormatRep() = default;
+
+void SparseCooFormatRep::InitIndices(const TensorShape& shape) {
+  auto ml_ind_type = DataTypeImpl::GetType<int64_t>();
+  void* data = IndicesStart(ml_ind_type->Size());
+  indices_ = Tensor(ml_ind_type, shape, data, Location());
+}
+
+Status SparseCooFormatRep::Copy(const DataTransferManager& data_transfer_manager, const AllocatorPtr& allocator,
```

Comment:
Copy(const DataTransferManager&, ...) and Copy(const IDataTransfer&, ...) seem very similar. is it possible to reduce duplication? could DataTransferManager implement IDataTransfer?